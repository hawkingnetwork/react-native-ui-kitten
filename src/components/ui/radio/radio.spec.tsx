import React from 'react';
import { TouchableOpacity } from 'react-native';
import {
  render,
  fireEvent,
  waitForElement,
  shallow,
  RenderAPI,
} from 'react-native-testing-library';
import { ReactTestInstance } from 'react-test-renderer';
import {
  ApplicationProvider,
  ApplicationProviderProps,
} from '@kitten/theme';
import {
  Radio,
  RadioComponent,
  RadioProps,
} from './radio.component';
import {
  mapping,
  theme,
} from '../support/tests';

const Mock = (props?: RadioProps): React.ReactElement<ApplicationProviderProps> => {
  return (
    <ApplicationProvider
      mapping={mapping}
      theme={theme}>
      <Radio {...props} />
    </ApplicationProvider>
  );
};

const renderComponent = (props?: RadioProps): RenderAPI => {
  return render(
    <Mock {...props}/>,
  );
};

describe('@radio: matches snapshot', () => {

  it('default', () => {
    const component: RenderAPI = renderComponent();
    const { output } = shallow(component.getByType(RadioComponent));

    expect(output).toMatchSnapshot();
  });

  it('checked', () => {
    const component: RenderAPI = renderComponent({ checked: true });
    const { output } = shallow(component.getByType(Radio));

    expect(output).toMatchSnapshot();
  });

  it('disabled', () => {
    const component: RenderAPI = renderComponent({ disabled: true });
    const { output } = shallow(component.getByType(Radio));

    expect(output).toMatchSnapshot();
  });

  it('checked disabled', () => {
    const component: RenderAPI = renderComponent({
      checked: true,
      disabled: true,
    });
    const { output } = shallow(component.getByType(Radio));

    expect(output).toMatchSnapshot();
  });

  it('active', async () => {
    const component: RenderAPI = renderComponent();

    fireEvent(component.getByType(TouchableOpacity), 'pressIn');

    const active: ReactTestInstance = await waitForElement(() => {
      return component.getByType(Radio);
    });
    const { output: activeOutput } = shallow(active);

    fireEvent(component.getByType(TouchableOpacity), 'pressOut');

    const inactive: ReactTestInstance = await waitForElement(() => {
      return component.getByType(Radio);
    });
    const { output: inactiveOutput } = shallow(inactive);

    expect(activeOutput).toMatchSnapshot();
    expect(inactiveOutput).toMatchSnapshot('default');
  });

  it('active checked', async () => {
    const component: RenderAPI = renderComponent({ checked: true });

    fireEvent(component.getByType(TouchableOpacity), 'pressIn');

    const active: ReactTestInstance = await waitForElement(() => {
      return component.getByType(Radio);
    });
    const { output: activeOutput } = shallow(active);

    fireEvent(component.getByType(TouchableOpacity), 'pressOut');

    const inactive: ReactTestInstance = await waitForElement(() => {
      return component.getByType(Radio);
    });
    const { output: inactiveOutput } = shallow(inactive);

    expect(activeOutput).toMatchSnapshot();
    expect(inactiveOutput).toMatchSnapshot('checked');
  });

  it('with text (styled)', () => {
    const component: RenderAPI = renderComponent({
      text: 'Place your text',
      textStyle: {
        fontSize: 22,
        lineHeight: 24,
      },
    });
    const { output } = shallow(component.getByType(Radio));

    expect(output).toMatchSnapshot();
  });

});

describe('@radio: component checks', () => {

  it('* emits onChange with correct args', () => {
    const onChange = jest.fn();

    const component: RenderAPI = renderComponent({
      checked: false,
      onChange: onChange,
    });

    fireEvent.press(component.getByType(TouchableOpacity));

    expect(onChange).toBeCalledWith(true);
  });

});
