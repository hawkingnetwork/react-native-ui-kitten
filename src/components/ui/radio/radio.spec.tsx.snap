// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@radio: matches snapshot active 1`] = `
<Wrapper
  forwardedRef={null}
/>
`;

exports[`@radio: matches snapshot active checked 1`] = `
<Wrapper
  checked={true}
  forwardedRef={null}
/>
`;

exports[`@radio: matches snapshot active checked: checked 1`] = `
<Wrapper
  checked={true}
  forwardedRef={null}
/>
`;

exports[`@radio: matches snapshot active: default 1`] = `
<Wrapper
  forwardedRef={null}
/>
`;

exports[`@radio: matches snapshot checked 1`] = `
<Wrapper
  checked={true}
  forwardedRef={null}
/>
`;

exports[`@radio: matches snapshot checked disabled 1`] = `
<Wrapper
  checked={true}
  disabled={true}
  forwardedRef={null}
/>
`;

exports[`@radio: matches snapshot default 1`] = `
<TouchableOpacity
  activeOpacity={1}
  appearance="default"
  dispatch={[Function]}
  hitSlop={
    Object {
      "bottom": 16,
      "left": 16,
      "right": 16,
      "top": 16,
    }
  }
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onPress={[Function]}
  onPressIn={[Function]}
  onPressOut={[Function]}
  status="basic"
  style={
    Array [
      Object {},
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      },
      undefined,
      undefined,
    ]
  }
  theme={
    Object {
      "background-alternative-color-1": "#222B45",
      "background-alternative-color-2": "#1A2138",
      "background-alternative-color-3": "#151A30",
      "background-alternative-color-4": "#101426",
      "background-basic-color-1": "#FFFFFF",
      "background-basic-color-2": "#F7F9FC",
      "background-basic-color-3": "#EDF1F7",
      "background-basic-color-4": "#E4E9F2",
      "background-primary-color-1": "#3366FF",
      "background-primary-color-2": "#274BDB",
      "background-primary-color-3": "#1A34B8",
      "background-primary-color-4": "#102694",
      "border-alternative-color-1": "#222B45",
      "border-alternative-color-2": "#1A2138",
      "border-alternative-color-3": "#151A30",
      "border-alternative-color-4": "#101426",
      "border-alternative-color-5": "#101426",
      "border-basic-color-1": "#FFFFFF",
      "border-basic-color-2": "#F7F9FC",
      "border-basic-color-3": "#EDF1F7",
      "border-basic-color-4": "#E4E9F2",
      "border-basic-color-5": "#C5CEE0",
      "border-danger-color-1": "#FF3D71",
      "border-danger-color-2": "#DB2C66",
      "border-danger-color-3": "#B81D5B",
      "border-danger-color-4": "#94124E",
      "border-danger-color-5": "#700940",
      "border-info-color-1": "#0095FF",
      "border-info-color-2": "#006FD6",
      "border-info-color-3": "#0057C2",
      "border-info-color-4": "#0041A8",
      "border-info-color-5": "#002885",
      "border-primary-color-1": "#3366FF",
      "border-primary-color-2": "#274BDB",
      "border-primary-color-3": "#1A34B8",
      "border-primary-color-4": "#102694",
      "border-primary-color-5": "#091C7A",
      "border-success-color-1": "#00E096",
      "border-success-color-2": "#00B383",
      "border-success-color-3": "#008F72",
      "border-success-color-4": "#007566",
      "border-success-color-5": "#00524C",
      "border-warning-color-1": "#FFAA00",
      "border-warning-color-2": "#DB8B00",
      "border-warning-color-3": "#B86E00",
      "border-warning-color-4": "#945400",
      "border-warning-color-5": "#703C00",
      "color-basic-100": "#FFFFFF",
      "color-basic-1000": "#151A30",
      "color-basic-1100": "#101426",
      "color-basic-200": "#F7F9FC",
      "color-basic-300": "#EDF1F7",
      "color-basic-400": "#E4E9F2",
      "color-basic-500": "#C5CEE0",
      "color-basic-600": "#8F9BB3",
      "color-basic-700": "#2E3A59",
      "color-basic-800": "#222B45",
      "color-basic-900": "#1A2138",
      "color-basic-active": "#E4E9F2",
      "color-basic-control-transparent-100": "rgba(255, 255, 255, 0.08)",
      "color-basic-control-transparent-200": "rgba(255, 255, 255, 0.16)",
      "color-basic-control-transparent-300": "rgba(255, 255, 255, 0.24)",
      "color-basic-control-transparent-400": "rgba(255, 255, 255, 0.32)",
      "color-basic-control-transparent-500": "rgba(255, 255, 255, 0.40)",
      "color-basic-control-transparent-600": "rgba(255, 255, 255, 0.48)",
      "color-basic-default": "#F7F9FC",
      "color-basic-disabled": "#EDF1F7",
      "color-basic-focus": "#C5CEE0",
      "color-basic-hover": "#F7F9FC",
      "color-basic-transparent-100": "rgba(143, 155, 179, 0.08)",
      "color-basic-transparent-200": "rgba(143, 155, 179, 0.16)",
      "color-basic-transparent-300": "rgba(143, 155, 179, 0.24)",
      "color-basic-transparent-400": "rgba(143, 155, 179, 0.32)",
      "color-basic-transparent-500": "rgba(143, 155, 179, 0.40)",
      "color-basic-transparent-600": "rgba(143, 155, 179, 0.48)",
      "color-danger-100": "#FFF2F2",
      "color-danger-200": "#FFD6D9",
      "color-danger-300": "#FFA8B4",
      "color-danger-400": "#FF708D",
      "color-danger-500": "#FF3D71",
      "color-danger-600": "#DB2C66",
      "color-danger-700": "#B81D5B",
      "color-danger-800": "#94124E",
      "color-danger-900": "#700940",
      "color-danger-active": "#DB2C66",
      "color-danger-default": "#FF3D71",
      "color-danger-disabled": "#FFA8B4",
      "color-danger-focus": "#B81D5B",
      "color-danger-hover": "#FF708D",
      "color-danger-transparent-100": "rgba(255, 61, 113, 0.08)",
      "color-danger-transparent-200": "rgba(255, 61, 113, 0.16)",
      "color-danger-transparent-300": "rgba(255, 61, 113, 0.24)",
      "color-danger-transparent-400": "rgba(255, 61, 113, 0.32)",
      "color-danger-transparent-500": "rgba(255, 61, 113, 0.40)",
      "color-danger-transparent-600": "rgba(255, 61, 113, 0.48)",
      "color-info-100": "#F2F8FF",
      "color-info-200": "#C7E2FF",
      "color-info-300": "#94CBFF",
      "color-info-400": "#42AAFF",
      "color-info-500": "#0095FF",
      "color-info-600": "#006FD6",
      "color-info-700": "#0057C2",
      "color-info-800": "#0041A8",
      "color-info-900": "#002885",
      "color-info-active": "#006FD6",
      "color-info-default": "#0095FF",
      "color-info-disabled": "#94CBFF",
      "color-info-focus": "#0057C2",
      "color-info-hover": "#42AAFF",
      "color-info-transparent-100": "rgba(0, 149, 255, 0.08)",
      "color-info-transparent-200": "rgba(0, 149, 255, 0.16)",
      "color-info-transparent-300": "rgba(0, 149, 255, 0.24)",
      "color-info-transparent-400": "rgba(0, 149, 255, 0.32)",
      "color-info-transparent-500": "rgba(0, 149, 255, 0.40)",
      "color-info-transparent-600": "rgba(0, 149, 255, 0.48)",
      "color-primary-100": "#F2F6FF",
      "color-primary-200": "#D9E4FF",
      "color-primary-300": "#A6C1FF",
      "color-primary-400": "#598BFF",
      "color-primary-500": "#3366FF",
      "color-primary-600": "#274BDB",
      "color-primary-700": "#1A34B8",
      "color-primary-800": "#102694",
      "color-primary-900": "#091C7A",
      "color-primary-active": "#274BDB",
      "color-primary-default": "#3366FF",
      "color-primary-disabled": "#D9E4FF",
      "color-primary-focus": "#1A34B8",
      "color-primary-hover": "#598BFF",
      "color-primary-transparent-100": "rgba(51, 102, 255, 0.08)",
      "color-primary-transparent-200": "rgba(51, 102, 255, 0.16)",
      "color-primary-transparent-300": "rgba(51, 102, 255, 0.24)",
      "color-primary-transparent-400": "rgba(51, 102, 255, 0.32)",
      "color-primary-transparent-500": "rgba(51, 102, 255, 0.40)",
      "color-primary-transparent-600": "rgba(51, 102, 255, 0.48)",
      "color-secondary-100": "#F7F2FF",
      "color-secondary-200": "#E0D1FF",
      "color-secondary-300": "#C5A8FF",
      "color-secondary-400": "#A375FF",
      "color-secondary-500": "#884DFF",
      "color-secondary-600": "#6A39DB",
      "color-secondary-700": "#4F28B8",
      "color-secondary-800": "#371B94",
      "color-secondary-900": "#26117A",
      "color-success-100": "#EDFFF3",
      "color-success-200": "#B3FFD6",
      "color-success-300": "#8CFAC7",
      "color-success-400": "#51F0B0",
      "color-success-500": "#00E096",
      "color-success-600": "#00B383",
      "color-success-700": "#008F72",
      "color-success-800": "#007566",
      "color-success-900": "#00524C",
      "color-success-active": "#00B383",
      "color-success-default": "#00E096",
      "color-success-disabled": "#B3FFD6",
      "color-success-focus": "#008F72",
      "color-success-hover": "#51F0B0",
      "color-success-transparent-100": "rgba(0, 224, 150, 0.08)",
      "color-success-transparent-200": "rgba(0, 224, 150, 0.16)",
      "color-success-transparent-300": "rgba(0, 224, 150, 0.24)",
      "color-success-transparent-400": "rgba(0, 224, 150, 0.32)",
      "color-success-transparent-500": "rgba(0, 224, 150, 0.40)",
      "color-success-transparent-600": "rgba(0, 224, 150, 0.48)",
      "color-tertiary-100": "#FFDED6",
      "color-tertiary-200": "#FFDED6",
      "color-tertiary-300": "#FFDED6",
      "color-tertiary-400": "#FFDED6",
      "color-tertiary-500": "#FFDED6",
      "color-tertiary-600": "#FFDED6",
      "color-tertiary-700": "#FFDED6",
      "color-tertiary-800": "#FFDED6",
      "color-tertiary-900": "#FFDED6",
      "color-warning-100": "#FFFDF2",
      "color-warning-200": "#FFF1C2",
      "color-warning-300": "#FFE59E",
      "color-warning-400": "#FFC94D",
      "color-warning-500": "#FFAA00",
      "color-warning-600": "#DB8B00",
      "color-warning-700": "#B86E00",
      "color-warning-800": "#945400",
      "color-warning-900": "#703C00",
      "color-warning-active": "#DB8B00",
      "color-warning-default": "#FFAA00",
      "color-warning-disabled": "#FFE59E",
      "color-warning-focus": "#B86E00",
      "color-warning-hover": "#FFC94D",
      "color-warning-transparent-100": "rgba(255, 170, 0, 0.08)",
      "color-warning-transparent-200": "rgba(255, 170, 0, 0.16)",
      "color-warning-transparent-300": "rgba(255, 170, 0, 0.24)",
      "color-warning-transparent-400": "rgba(255, 170, 0, 0.32)",
      "color-warning-transparent-500": "rgba(255, 170, 0, 0.40)",
      "color-warning-transparent-600": "rgba(255, 170, 0, 0.48)",
      "icon-active-color": "#3366FF",
      "icon-basic-color": "#2E3A59",
      "icon-control-color": "#FFFFFF",
      "icon-disabled-color": "#E4E9F2",
      "icon-hint-color": "#C5CEE0",
      "outline-color": "#E4E9F2",
      "text-alternate-color": "#FFFFFF",
      "text-basic-color": "#1A2138",
      "text-control-color": "#FFFFFF",
      "text-danger-active-color": "#DB2C66",
      "text-danger-color": "#FF3D71",
      "text-danger-disabled-color": "#FFA8B4",
      "text-danger-focus-color": "#B81D5B",
      "text-danger-hover-color": "#FF708D",
      "text-disabled-color": "#C5CEE0",
      "text-hint-color": "#8F9BB3",
      "text-info-active-color": "#006FD6",
      "text-info-color": "#0095FF",
      "text-info-disabled-color": "#94CBFF",
      "text-info-focus-color": "#0057C2",
      "text-info-hover-color": "#42AAFF",
      "text-primary-active-color": "#274BDB",
      "text-primary-color": "#3366FF",
      "text-primary-disabled-color": "#D9E4FF",
      "text-primary-focus-color": "#1A34B8",
      "text-primary-hover-color": "#598BFF",
      "text-success-active-color": "#00B383",
      "text-success-color": "#00E096",
      "text-success-disabled-color": "#B3FFD6",
      "text-success-focus-color": "#008F72",
      "text-success-hover-color": "#51F0B0",
      "text-warning-active-color": "#DB8B00",
      "text-warning-color": "#FFAA00",
      "text-warning-disabled-color": "#FFE59E",
      "text-warning-focus-color": "#B86E00",
      "text-warning-hover-color": "#FFC94D",
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          },
          Object {
            "position": "absolute",
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#E4E9F2",
            "borderColor": "#C5CEE0",
            "borderRadius": 12,
            "borderWidth": 1,
            "height": 24,
            "width": 24,
          },
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderRadius": 8,
              "height": 16,
              "width": 16,
            },
            Object {},
          ]
        }
      />
    </View>
  </View>
</TouchableOpacity>
`;

exports[`@radio: matches snapshot disabled 1`] = `
<Wrapper
  disabled={true}
  forwardedRef={null}
/>
`;

exports[`@radio: matches snapshot with text (styled) 1`] = `
<Wrapper
  forwardedRef={null}
  text="Place your text"
  textStyle={
    Object {
      "fontSize": 22,
      "lineHeight": 24,
    }
  }
/>
`;
